src/cli.py

```diff
@@ -42,479 +104,1158 @@ def _format_action_invocation(self, action):
-            )
-            assert 'settings' in d
-        except Exception:
-            pass # better error handling?
-        return d
-
-    # get list of pods
-    if not pod_list:
-        pod_list = os.listdir(os.path.join(code_root, _pod_dir))
-        pod_list = [s for s in pod_list if not s.startswith(('_','.'))]
-        pod_list.sort(key=six.text_type.lower)
-    if pod == 'list':
-        return pod_list
-
-    # load JSON files
-    if not pod:
-        # load all of them
-        pods = dict()
-        realm_list = set()
-        bad_pods = []
-        realms = collections.defaultdict(list)
-        for p in pod_list:
-            d = _load_one_json(p)
-            if not d:
-                bad_pods.append(p)
-                continue
-            pods[p] = d
-            # PODs requiring data from multiple realms get stored in the dict
-            # under a tuple of those realms; realms stored indivudally in realm_list
-            _realm = util.coerce_to_iter(d['settings'].get('realm', None), tuple)
-            if len(_realm) == 0:
-                continue
-            elif len(_realm) == 1:
-                _realm = _realm[0]
-                realm_list.add(_realm)
-            else:
-                realm_list.update(_realm)
-            realms[_realm].append(p)
-        for p in bad_pods:
-            pod_list.remove(p)
-        return PodDataTuple(
-            pod_data=pods, realm_data=realms,
-            sorted_lists={
-                "pods": pod_list,
-                "realms": sorted(list(realm_list), key=six.text_type.lower)
-            }
-        )
-    else:
-        if pod not in pod_list:
-            print("Couldn't recognize POD {} out of the following diagnostics:".format(pod))
-            print(', '.join(pod_list))
-            return dict()
-        return _load_one_json(pod)
+    def init_site(self):
+        """We allow site-specific installations to customize the CLI, so before 
+        we construct the CLI parser we need to determine what site to use. We do
+        this by running a parser that only looks for the ``--site`` flag.
 
+        This sets the ``site`` attribute and populates the global and
+        site-specific settings dicts.
+        """
+        config = CLIConfigManager()
+        config.read_defaults(DefaultsFileTypes.GLOBAL)
+        default_site = config.partial_defaults.get('site', config.default_site)
 
-class InfoCLIHandler(object):
-    def __init__(self, code_root, arg_list):
-        def _add_topic_handler(keywords, function):
-            # keep cmd_list ordered
-            keywords = util.coerce_to_iter(keywords)
-            self.cmd_list.extend(keywords)
-            for k in keywords:
-                self.cmds[k] = function
+        self.sites = [d for d in os.listdir(config.sites_dir) \
+            if os.path.isdir(os.path.join(config.sites_dir, d)) \
+                and not d.startswith(('.','_'))]
+        # if default_site in self.sites:
```

Comment:
Delete commented-out lines?

Reply:
Done in NOAA-GFDL/MDTF-diagnostics@7b47104f5077872e6b20e1af63fb487da281db9b